### PR TITLE
Add __resolveType to _Entity union

### PR DESCRIPTION
**Pull request by michael-watson**
This fixes issue #1772

### CODE REVIEW

**Comment by trevor-scheer** [comment]
@michael-watson can we close out https://github.com/apollographql/federation/issues/411 and https://github.com/apollographql/federation/issues/348 with this PR?

**Review by trevor-scheer — APPROVED** [review]
Would you add a changelog entry to the `subgraph-js/CHANGELOG.md` please?

File: subgraph-js/CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 This CHANGELOG pertains only to Apollo Federation packages in the 2.x range. The Federation v0.x equivalent for this package can be found [here](https://github.com/apollographql/federation/blob/version-0.x/subgraph-js/CHANGELOG.md) on the `version-0.x` branch of this repo.
 
+- Add __resolveType to _Entity union
```
Comment (trevor-scheer) [inline subgraph-js/CHANGELOG.md:5]:
```suggestion
- Add __resolveType to _Entity union [PR #1773](https://github.com/apollographql/federation/pull/1773)
```

**Comment by michael-watson** [comment]
Yes, I believe we can close #411 and #348. The default mock options in `ApolloServer` will work for `mocks: true, mockEntireSchema: false`. [It](https://github.com/apollographql/apollo-server/blob/main/packages/apollo-server-core/src/ApolloServer.ts#L661) translates directly to `addMocksToSchema` with the `preserveResolver` options.

**Pull request merged**